src/components/embedded/EmbeddedMessage.tsx

```diff
@@ -15,19 +15,40 @@
  ***************************************************************************** */
 
 import React, { useEffect, useState } from 'react';
-import { EventMessage, MessageViewType } from '../../models/EventMessage';
+import { EventMessage, MessageViewType, EventMessageItem } from '../../models/EventMessage';
 import MessageCardBase from '../message/message-card/MessageCardBase';
 import SplashScreen from '../SplashScreen';
 
 function EmbeddedMessage({ messageId }: { messageId: string }) {
 	const [message, setMessage] = useState<EventMessage | null>();
+	const [messageList, setMessageList] = React.useState<EventMessageItem[]>([]);
 	const [viewType, setViewType] = useState(MessageViewType.JSON);
 	const [errorStatus, setErrorStatus] = useState<string | null>(null);
 
 	useEffect(() => {
 		getMessage();
 	}, []);
 
+	useEffect(() => {
+		if (message)
```

Comment:
let's use braces {}

Does that code even work ? it looks like it should cause infinite loop. Effect has no dependencies and you changing state inside